Documentation/Evolution/StringProcessingAlgorithms.md

```diff
@@ -0,0 +1,545 @@
+# String processing algorithms
+
+## Introduction
+
+The Swift standard library's string processing algorithms are underpowered compared to other popular programming and scripting languages. Some of these omissions can be papered over through bridged `NSString` API available through the Foundation framework, but these are still incomplete and bring in linkage requirements to Foundation.
+
+We propose:
+
+1. New regex-powered algorithms over strings, bringing the standard library up to parity with scripting languages
+2. Generic `Collection` equivalents of these algorithms in terms of subsequences
+3. `protocol CustomRegexComponent`, allowing libraries to vend types that can be intermixed as components of regexes
+
+This proposal is part of a larger [regex-powered string processing initiative](https://forums.swift.org/t/declarative-string-processing-overview/52459). Throughout the document, we will reference the still-in-progress [`RegexProtocol`, `Regex`](https://github.com/apple/swift-experimental-string-processing/blob/main/Documentation/Evolution/StronglyTypedCaptures.md), and [result builder DSL](https://forums.swift.org/t/pitch-regular-expression-literals/52820), but these are in flux and not formally part of this proposal. Further discussion of regex specifics is out of scope of this proposal and better discussed in another thread (see [Pitch and Proposal Status](https://github.com/apple/swift-experimental-string-processing/issues/107) for links to relevant threads).
```

Comment:
The "result builder DSL" is linking to the wrong pitch though, the link points to the regex literal pitch. We don't have a DSL pitch yet — I think you can just use plain text for now, or link to https://github.com/rxwei/swift-experimental-string-processing/blob/dsl-api-walkthrough/Documentation/DSLAPIWalkthough.md (which I plan to turn into a pitch very soon).

Reply:
Thanks for catching that. I removed the link all together.